packages/framework-provider-azure-infrastructure/src/infrastructure/utils.ts

```diff
@@ -45,13 +45,13 @@ export async function buildResource(
 }
 
 export async function packageAzureFunction(functionDefinitions: Array<FunctionDefinition>): Promise<any> {
-  const output = fs.createWriteStream(path.join(os.tmpdir(), 'example.zip'))
+  const output = fs.createWriteStream(path.join(os.tmpdir(), 'functionApp.zip'))
   const archive = archiver('zip', {
     zlib: { level: 9 }, // Sets the compression level.
   })
 
   archive.pipe(output)
-  archive.glob('**/*')
```

Comment:
This was the hardest issue to solve. 
The deployment were working sometimes. The times it failed the errors were different. IT was really weird. It turns out that the zip function deployment to azure storage was not ending because not all the packages in the nodes modules were copied (38 MB) We don't need to copy everything only what is in the .deploy folder. The zip got smaller and we got a bunch of successful deployments consistently now.